test/mongoGetCommand.test.ts

```diff
@@ -319,21 +333,86 @@ suite("scrapbook parsing Tests", () => {
+        assert.deepEqual(command.argumentObjects, [{ name: false }]);
+    });
+    test("test function call token inside quotes", () => {
+        let text = `db.test1.insertMany({'name': 'false'})`;
+        let command = getCommandFromText(text, new Position(0, 0));
+        assert.deepEqual(command.argumentObjects, [{ name: "false" }]);
+    });
+    test("test function call with an empty string property value", () => {
+        let text = `db.test1.insertMany({'name': ''})`;
+        let command = getCommandFromText(text, new Position(0, 0));
+        assert.deepEqual(command.argumentObjects, [{ name: "" }]);
+    });
+    test("test function call with array and multiple arguments", () => {
+        let text = `db.test1.find({'roles': ['readWrite', 'dbAdmin']}, {'resources': ['secondary', 'primary']})`;
+        let command = getCommandFromText(text, new Position(0, 0));
+        assert.deepEqual(command.argumentObjects, [{ roles: ["readWrite", "dbAdmin"] }, { resources: ["secondary", "primary"] }]);
+    });
+    test("test function call with nested objects", () => {
+        let text = `db.test1.find({'roles': [{'optional': 'yes'}]}, {'resources': ['secondary', 'primary']})`;
+        let command = getCommandFromText(text, new Position(0, 0));
+        assert.deepEqual(command.argumentObjects, [{ roles: [{ optional: "yes" }] }, { resources: ["secondary", "primary"] }]);
+    });
+
+    // Single quotes - intermediate states to replicate typing into the console
+    test("test incomplete function call - replicate user typing - missing propertyValue", () => {
+        let text = `db.test1.find({"name": {"First" : } })`;
+        let command = getCommandFromText(text, new Position(0, 0));
+        assert.deepEqual(command.argumentObjects, [{ name: { First: {} } }]);
+    });
+
+    test("test incomplete function call - replicate user typing - missing colon & propertyValue", () => {
+        let text = `db.test1.find({"name": {"First"  } })`;
+        let command = getCommandFromText(text, new Position(0, 0));
+        assert.deepEqual(command.argumentObjects, [{ name: { First: {} } }]);
+    });
+
+    //This test will fail. See https://github.com/Microsoft/vscode-cosmosdb/issues/689
+    // test("test incomplete function call - replicate user typing - no function call yet", () => {
+    //     let text = `db.test1.`;
+    //     let command = getCommandFromText(text, new Position(0, 0));
+    //     assert.deepEqual(command.argumentObjects, undefined);
+    //     assert.deepEqual(command.collection, "test1");
+    // });
+
+    /* This test will fail. Unquoted strings need to be fixed.
+    test("test function call with no quotes", () => {
+        let arg0 = `{name: 'First'}`;
+        let text = `db.test1.insertMany(${arg0})`;
+        let command = getCommandFromText(text, new Position(0, 0));
+        const argument = command.argumentObjects[0];
+        assert.deepEqual(argument, { name: "First" });
+    });
+    */
+
```

Comment:
Please test these strings:

"That's all"
'That"s all'
"Hello \"there\""
'Hello \'there\''